pandas/tests/io/test_sql.py

```diff
@@ -622,9 +628,15 @@ def test_date_parsing(self):
 
         df = sql.read_sql_query("SELECT * FROM types_test_data", self.conn,
                                 parse_dates={'IntDateCol': 's'})
-
         assert issubclass(df.IntDateCol.dtype.type, np.datetime64)
 
+        df = sql.read_sql_query("SELECT * FROM types_test_data", self.conn,
+                                parse_dates={'IntDateOnlyCol': '%Y%m%d'})
+        assert issubclass(df.IntDateOnlyCol.dtype.type, np.datetime64)
```

Comment:
@jreback I included a test. I also noticed that the existing tests are not enough. It might be that the parsing returns a `NaT` which satisfies the classing condition but the values are wrong. This is the reason I added an explicit test that checks that the resulting *values* are correct. As a matter of fact while implementing the improvement suggested by @jorisvandenbossche the tests passed but the returned values were `NaT`s (the reason was a missing `,` in the code @jorisvandenbossche  suggested)

Would you agree there's a problem with the tests?

Reply:
Yes, those are indeed not very thorough. Do you want to add such a similar check to the others as well?

Reply:
@jorisvandenbossche Within this same PR?

Reply:
I think that would be good yes (because as you said, with your changes you could actually silently 'break' them now)

Reply:
@jorisvandenbossche I have included more explicit tests.